docs/tutorials/data_manipulation/pandasdataframes.md.input

```diff
@@ -354,3 +355,62 @@ for index, ax in zip(indices, axes):
     forecasts[index].plot(prediction_intervals=[90.], color='g')
     plt.legend(["observed", "predicted median", "predicted 90% quantile"])
 ```
+
+
+## Use case 2 - Loading data from a single "long" CSV file and filling missing rows
+
+In this section we load data from a long-dataframe format, where each time series is
+stacked on top of each other and has gaps that need to be filled.
+
+To fill the gaps (to include new rows), we group by the `item_id` and reindex the
+dataframe according to our needs. New rows will be created with `NaN`s which
+we fill with -1.
+Note, we go through this procedure because we are including new rows. If your
+csv already contains all the rows and has `NaN`s as missing values, you can simply do
+`df = pd.read_csv("file.csv").fillna(value)` and use the
+`PandasDataset.from_long_dataframe(df, item_id="item_id")` constructor.
+
+```python
+import pandas as pd
+from gluonts.dataset.pandas import PandasDataset
+
+url = (
+    "https://gist.githubusercontent.com/rsnirwan/91508607f7f8778913012afc87d57538"
+    "/raw/8b4883b2f1056f3e5e4a63e7f0daf21366a2eb49/dummy_ts_with_gaps.csv"
+)
+df = pd.read_csv(url, index_col=0, parse_dates=True)
+print(df)
+
+# get the latest date. we also fill forward if latest entries are missing.
+max_end = max(df.groupby("item_id").apply(lambda _df: _df.index[-1]))
+dfs_dict = {}
+for item_id, gdf in df.groupby("item_id"):
+    new_index = pd.date_range(gdf.index[0], end=max_end, freq="1D")
+    dfs_dict[item_id] = gdf.reindex(new_index).drop("item_id", axis=1).fillna(-1)
+
+ds = PandasDataset(dfs_dict)
+for ts in ds:
+    print(ts)
+```
+
+## Use case 3 - Loading data from a `wide`-DataFrame
+
+In case a `dataframe` is given in a `wide` instead of `long` format,
+where time series are stacked side-by-side instead of on top of each other,
+we can easily pivot from wide to long using `pd.melt`.
+
+```python
+import numpy as np
+import pandas as pd
+from gluonts.dataset.pandas import PandasDataset
+
+N, T = 3, 10
+index = pd.date_range("2021-01-01", freq="1D", periods=T)
+wide = pd.DataFrame(np.random.normal(size=(T, N)), columns=["A", "B", "C"], index=index)
```

Comment:
Also here I would just load a CSV from somewhere, since that's how one would want to use it